docs/task_overview/introduction.md

```diff
@@ -0,0 +1,104 @@
+
+- {py:class}`~columnflow.tasks.reduction.ReduceEvents`: Task to apply the masks created in
+{py:class}`~columnflow.tasks.selection.SelectEvents` on the datasets. All
+tasks below ReduceEvents in the task graph use the parquet
+file resulting from ReduceEvents to work on, not the
+original dataset. The columns to be conserved after
+ReduceEvents are to be given in the analysis config under
+the ```config.x.keep_columns``` argument in a ```DotDict``` structure
+(from {py:mod}`~columnflow.util`).
+
+- {py:class}`~columnflow.tasks.production.ProduceColumns`: Task to produce additional columns for
+the reduced datasets, e.g. for new high level variables. This task uses objects of the
+{py:class}`~columnflow.production.Producer` class to create the new columns. The new columns are
+saved in a parquet file that can be used by the task below on the task graph. The argument
+```--producer``` followed by the name of the Producer object
+to be run is needed for this task to run. A default value for this argument can be set in the
+analysis config.
+
+- {py:class}`~columnflow.tasks.ml.PrepareMLEvents`, {py:class}`~columnflow.tasks.ml.MLTraining`,
+{py:class}`~columnflow.tasks.ml.MLEvaluation`: Tasks to
+train, evaluate neural networks and plot (to be implemented) their results.
+TODO: more informations? output type?
+all tf based? or very general and almost everything must be set in the training scripts?
+and if general, what is taken care of?
+
+- {py:class}`~columnflow.tasks.histograms.CreateHistograms`: Task to create histograms with the
+python package [Hist](https://hist.readthedocs.io/en/latest/) which can be used by the tasks below
+in the task graph. From this task on, the ```--producer``` argument is replaced by
+```--producers```. The histograms are saved in a pickle file.
+TODO: more informations?
+
+- {py:class}`~columnflow.tasks.cms.inference.CreateDatacards`: TODO
+
+- ```Merge``` tasks (e.g. {py:class}`~columnflow.tasks.reduction.MergeReducedEvents`,
+{py:class}`~columnflow.tasks.histograms.MergeHistograms`): Tasks to merge the local outputs from
+the various occurences of the corresponding tasks. TODO: details? why needed? Only convenience?
+Or I/O?
+
+There are also CMS-specialized tasks, like
+{py:class}`~columnflow.tasks.cms.external.CreatePileUpWeights`, which are described in the
+{ref}`CMS specializations section <cms_specializations_section>`. As a note, the CreatePileUpWeights task is
+interesting from a workflow point of view as it is an example of a task required through an object
+of the {py:class}`~columnflow.production.Producer` class. This behaviour can be observed in the
+{py:meth}`~columnflow.production.cms.pileup.pu_weight_requires` method.
+
+TODO: maybe interesting to have examples e.g. for the usage of the
+parameters for the 2d plots. Maybe in the example section, or after the next subsection, such that
+all parameters are explained? If so, at least to be mentioned here.
```

Comment:
I think we can just add another two sentences and a link to the new plotting user guide

Reply:
Plotting is currently completely missing from this list in the first place, so I guess the comment can't be applied easily atm. Still, I think it would be good to include it already here